### PR TITLE
refactor: pull out credential type label function

### PR DESCRIPTION
**Pull request by bodnarbm**
Summary
=======

As prep work for some algorithm improvements to labeling the
credential type for each program, this pulls out the credential
type labeling code into its own function that operates per
dataframe row using the apply method. This will hopefully also
improve readability of which labels trump others when multiple
matches are possible through using early returns.

This refactor should not change the existing results.

Test Plan
=========

I ran the merge-credentials step on the base branch to get a baseline of the existing merged file and saved that a copy of that file. Then I ran merge-credentials for this branch and compared the two files using diff, ensuring there was no change to output caused by this refactor.

### CODE REVIEW

File: backend/data/program-credentials/merge-credentials.py [hunk 1]
```diff
@@ -32,58 +32,47 @@ def merge_lookup_label(result: pd.DataFrame, lookup_df: pd.DataFrame, column: st
...
-    # Apprenticeship
-    df.loc[(df['OFFICIALNAME'].str.contains('Apprentice')) & (
-                df['CREDENTIALTYPE'] != 'Pre-Apprenticeship'), 'CREDENTIALTYPE'] = 'Apprenticeship'
+    if 'A.S.' in official_name:
+        return 'Associate Degree'
 
-    # Secondary School Diploma
-    df.loc[df['OFFICIALNAME'].str.contains(
-        'High School Diploma - Adults'), 'CREDENTIALTYPE'] = 'Secondary School Diploma'
+    ESL_TOKENS = { 'ESL', 'English as a Second Language'}
```
Comment (derekkinsman) [inline backend/data/program-credentials/merge-credentials.py:55]: Should this be `ESL_NAME_TOKENS` to be consistent with next token?

**Review by derekkinsman — COMMENTED** [review]
Left a few inline questions about naming conventions.

File: backend/data/program-credentials/merge-credentials.py (hunk 1 above)
Comment (bodnarbm) [reply backend/data/program-credentials/merge-credentials.py]: 👍 Fixed up ESL_TOKENS to ESL_NAME_TOKENS.

File: backend/data/program-credentials/merge-credentials.py [hunk 2]
```diff
@@ -32,58 +32,47 @@ def merge_lookup_label(result: pd.DataFrame, lookup_df: pd.DataFrame, column: st
...
-    # Masters
-    df.loc[(df['OFFICIALNAME'].str.contains('M.', regex=False)) & (
-        df['DEGREEAWARDEDNAME'].str.contains('Master')), 'CREDENTIALTYPE'] = 'Masters Degree'
+    if 'Pre-Apprentice' in official_name:
+        return 'Pre-Apprenticeship'
 
-    # License
-    df.loc[(df['LEADTOLICENSE'] == "1") & (
-                df['LEADTOINDUSTRYCREDENTIAL'] == "0"), 'CREDENTIALTYPE'] = 'License'
+    GED_NAME_TOKENS = { 'GED', 'General Education Diploma', 'High School Equivalence', 'High School Equivalency'}
```
[the remark this answers is not in the document]
Comment (bodnarbm) [reply backend/data/program-credentials/merge-credentials.py:68]: Fixed up ESL_NAME_TOKENS to match instead.

**Review by derekkinsman — APPROVED** [review]
Looks good.

**Review by namanaman — APPROVED** [review]
awesome

**Review by namanaman — APPROVED** [review]
awesome

**Pull request merged**